Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -43,22 +50,80 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSCache alloc] init];
         self.sectionControllersBySection.countLimit = [WMFExploreSection totalMaxNumberOfSections];
-        self.sectionsBySectionController            = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                                                            valueOptions:NSMapTableWeakMemory];
+
+        /*
+         Controllers (keys in the map) are retained by the NSCache and can be evicted at any time. On the other hand,
+         sections (values in the map) are "value types" which should be owned by the receiver.  In other words, sections
+         are created as needed, and not retained for any particular reason, and it would be unintentional for a value in
```

Comment:
> I can't think of a reason that we would want the map table to retain a section if it has been released everywhere else.

This is mostly pedantic, since the cache retains (note, **not** copies) the sections as its keys, but I should be able to retrieve a controller for a _copy_ of the section of the section I previously associated it with. IOW, it shouldn't matter that it's a copy of the _original_ key as long as they're equivalent.  On the other hand, I can't think of any cases—pedantic or practical—that I would want sections to be released in the reverse lookup map.  As long as the controller is in the cache, I should be able to perform a reverse lookup of the section it's associated with, **regardless of whether that specific section object was dereferenced**.

Put in Swift terms, think of controllers as classes & sections as structs.  You can't dereference a struct because you can't reference it in the first place.  They're "value types," meaning they only represent a specific value, like a number, not a specific object in memory.